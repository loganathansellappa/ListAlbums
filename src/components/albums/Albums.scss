.albums-container {
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: 10px;
  .search {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  .searchInput {
    width: 100%;
    background-image: linear-gradient(0deg, red 0%, blue 100%);
    color: #ffffff;
    font-size: 2rem;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
  }
  .searchHint {
    font-weight: bold;
  }

  .albums {
    display: flex;
    flex-wrap: wrap;
    padding-left: 10%;
    padding-top: 1%;
    padding-right: 10%;
    justify-content: center;
  }
}
