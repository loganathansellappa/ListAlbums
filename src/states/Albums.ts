/**
 * State Management
 * Simple actions to add/remove album in favorite list
 */
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Entry } from '../@types/Album';

export interface AlbumsState {
  favourites: Array<Entry> | [];
}

export const initialState: AlbumsState = {
  favourites: [],
};

export const albumsSlice = createSlice({
  name: 'albums',
  initialState,
  reducers: {
    add: (state, action: PayloadAction<Entry>) => {
      state.favourites = [...state.favourites, action.payload];
    },
    remove: (state, action: PayloadAction<string>) => {
      state.favourites = state.favourites.filter(
        (album) => action.payload !== album.id.attributes['im:id']
      );
    },
  },
});

export const { add, remove } = albumsSlice.actions;

export default albumsSlice.reducer;
