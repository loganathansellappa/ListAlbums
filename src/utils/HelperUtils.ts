/**
 * Utils functions
 */
import { Entry } from '../@types/Album';

export const apiService = () => `${process.env.MUSIC_URL}`;
export const dataLimit = () => `${process.env.DEFAULT_LIMIT}`;
export const filteredAlbums = (albums: Array<Entry>, typeHead: string) =>
  albums?.filter(
    (album: Entry) =>
      album.title.label.toLowerCase().includes(typeHead) ||
      album['im:name'].label.toLowerCase().includes(typeHead)
  );
