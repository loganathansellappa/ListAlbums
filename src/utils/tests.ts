import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};
export const testStore = () => {
  const mockStore = configureStore();
  const initialState = { favoriteAlbums: { favourites: [] } };
  const store = mockStore(initialState);
  return store;
};

export const mockResource = () => {
  return {
    feed: {
      author: {
        name: { label: 'iTunes Store' },
        uri: { label: 'http://www.apple.com/itunes/' },
      },
      entry: [
        {
          'im:name': { label: 'Life After Death' },
          'im:image': [
            {
              label:
                'https://is2-ssl.mzstatic.com/image/thumb/Music112/v4/5d/1b/af/5d1baf4a-496c-be36-b5c1-ba5bcea1c2c0/22UMGIM57228.rgb.jpg/55x55bb.png',
              attributes: { height: '55' },
            },
            {
              label:
                'https://is4-ssl.mzstatic.com/image/thumb/Music112/v4/5d/1b/af/5d1baf4a-496c-be36-b5c1-ba5bcea1c2c0/22UMGIM57228.rgb.jpg/60x60bb.png',
              attributes: { height: '60' },
            },
            {
              label:
                'https://is2-ssl.mzstatic.com/image/thumb/Music112/v4/5d/1b/af/5d1baf4a-496c-be36-b5c1-ba5bcea1c2c0/22UMGIM57228.rgb.jpg/170x170bb.png',
              attributes: { height: '170' },
            },
          ],
          'im:itemCount': { label: '15' },
          'im:price': {
            label: '$10.99',
            attributes: { amount: '10.99', currency: 'USD' },
          },
          'im:contentType': {
            'im:contentType': { attributes: { term: 'Album', label: 'Album' } },
            attributes: { term: 'Music', label: 'Music' },
          },
          rights: {
            label:
              'ForeFront Records; ℗ 2022 F.L. Inc., under exclusive license to Capitol CMG, Inc.',
          },
          title: { label: 'Life After Death - TobyMac' },
          link: {
            attributes: {
              rel: 'alternate',
              type: 'text/html',
              href: 'https://music.apple.com/us/album/life-after-death/1626343519?uo=2',
            },
          },
          id: {
            label:
              'https://music.apple.com/us/album/life-after-death/1626343519?uo=2',
            attributes: { 'im:id': '1626343519' },
          },
          'im:artist': {
            label: 'TobyMac',
            attributes: {
              href: 'https://music.apple.com/us/artist/tobymac/635550?uo=2',
            },
          },
          category: {
            attributes: {
              'im:id': '22',
              term: 'Christian & Gospel',
              scheme:
                'https://music.apple.com/us/genre/music-christian/id22?uo=2',
              label: 'Christian',
            },
          },
          'im:releaseDate': {
            label: '2022-08-19T00:00:00-07:00',
            attributes: { label: 'August 19, 2022' },
          },
        },
        {
          'im:name': { label: 'Viva Las Vengeance' },
          'im:image': [
            {
              label:
                'https://is3-ssl.mzstatic.com/image/thumb/Music112/v4/bc/49/c6/bc49c6b9-9a12-dd6c-ac50-e4409cf8a094/075679762108.jpg/55x55bb.png',
              attributes: { height: '55' },
            },
            {
              label:
                'https://is5-ssl.mzstatic.com/image/thumb/Music112/v4/bc/49/c6/bc49c6b9-9a12-dd6c-ac50-e4409cf8a094/075679762108.jpg/60x60bb.png',
              attributes: { height: '60' },
            },
            {
              label:
                'https://is5-ssl.mzstatic.com/image/thumb/Music112/v4/bc/49/c6/bc49c6b9-9a12-dd6c-ac50-e4409cf8a094/075679762108.jpg/170x170bb.png',
              attributes: { height: '170' },
            },
          ],
          'im:itemCount': { label: '12' },
          'im:price': {
            label: '$9.99',
            attributes: { amount: '9.99', currency: 'USD' },
          },
          'im:contentType': {
            'im:contentType': { attributes: { term: 'Album', label: 'Album' } },
            attributes: { term: 'Music', label: 'Music' },
          },
          rights: {
            label:
              '℗ 2022 Fueled By Ramen LLC for the United States and WEA International Inc. for the world outside of the United States.  A Warner Music Group Company',
          },
          title: { label: 'Viva Las Vengeance - Panic! At the Disco' },
          link: {
            attributes: {
              rel: 'alternate',
              type: 'text/html',
              href: 'https://music.apple.com/us/album/viva-las-vengeance/1626561623?uo=2',
            },
          },
          id: {
            label:
              'https://music.apple.com/us/album/viva-las-vengeance/1626561623?uo=2',
            attributes: { 'im:id': '1626561623' },
          },
          'im:artist': {
            label: 'Panic! At the Disco',
            attributes: {
              href: 'https://music.apple.com/us/artist/panic-at-the-disco/80456331?uo=2',
            },
          },
          category: {
            attributes: {
              'im:id': '20',
              term: 'Alternative',
              scheme:
                'https://music.apple.com/us/genre/music-alternative/id20?uo=2',
              label: 'Alternative',
            },
          },
          'im:releaseDate': {
            label: '2022-08-19T00:00:00-07:00',
            attributes: { label: 'August 19, 2022' },
          },
        },
        {
          'im:name': {
            label: 'Finally Enough Love: 50 Number Ones (2022 Remasters)',
          },
          'im:image': [
            {
              label:
                'https://is1-ssl.mzstatic.com/image/thumb/Music126/v4/f8/70/c2/f870c27d-81f3-08b7-f66d-b7dc3ffe6a1f/603497839797.jpg/55x55bb.png',
              attributes: { height: '55' },
            },
            {
              label:
                'https://is5-ssl.mzstatic.com/image/thumb/Music126/v4/f8/70/c2/f870c27d-81f3-08b7-f66d-b7dc3ffe6a1f/603497839797.jpg/60x60bb.png',
              attributes: { height: '60' },
            },
            {
              label:
                'https://is1-ssl.mzstatic.com/image/thumb/Music126/v4/f8/70/c2/f870c27d-81f3-08b7-f66d-b7dc3ffe6a1f/603497839797.jpg/170x170bb.png',
              attributes: { height: '170' },
            },
          ],
          'im:itemCount': { label: '50' },
          'im:price': {
            label: '$24.99',
            attributes: { amount: '24.99', currency: 'USD' },
          },
          'im:contentType': {
            'im:contentType': { attributes: { term: 'Album', label: 'Album' } },
            attributes: { term: 'Music', label: 'Music' },
          },
          rights: { label: '℗ 2022 Warner Records LLC.' },
          title: {
            label:
              'Finally Enough Love: 50 Number Ones (2022 Remasters) - Madonna',
          },
          link: {
            attributes: {
              rel: 'alternate',
              type: 'text/html',
              href: 'https://music.apple.com/us/album/finally-enough-love-50-number-ones-2022-remasters/1622113557?uo=2',
            },
          },
          id: {
            label:
              'https://music.apple.com/us/album/finally-enough-love-50-number-ones-2022-remasters/1622113557?uo=2',
            attributes: { 'im:id': '1622113557' },
          },
          'im:artist': {
            label: 'Madonna',
            attributes: {
              href: 'https://music.apple.com/us/artist/madonna/20044?uo=2',
            },
          },
          category: {
            attributes: {
              'im:id': '14',
              term: 'Pop',
              scheme: 'https://music.apple.com/us/genre/music-pop/id14?uo=2',
              label: 'Pop',
            },
          },
          'im:releaseDate': {
            label: '2022-08-19T00:00:00-07:00',
            attributes: { label: 'August 19, 2022' },
          },
        },
        {
          'im:name': { label: '& - EP' },
          'im:image': [
            {
              label:
                'https://is4-ssl.mzstatic.com/image/thumb/Music126/v4/88/ba/eb/88baebfa-1338-f4c4-e9ab-a86877a2b763/20UMGIM88356.rgb.jpg/55x55bb.png',
              attributes: { height: '55' },
            },
            {
              label:
                'https://is1-ssl.mzstatic.com/image/thumb/Music126/v4/88/ba/eb/88baebfa-1338-f4c4-e9ab-a86877a2b763/20UMGIM88356.rgb.jpg/60x60bb.png',
              attributes: { height: '60' },
            },
            {
              label:
                'https://is4-ssl.mzstatic.com/image/thumb/Music126/v4/88/ba/eb/88baebfa-1338-f4c4-e9ab-a86877a2b763/20UMGIM88356.rgb.jpg/170x170bb.png',
              attributes: { height: '170' },
            },
          ],
          'im:itemCount': { label: '6' },
          'im:price': {
            label: '$6.99',
            attributes: { amount: '6.99', currency: 'USD' },
          },
          'im:contentType': {
            'im:contentType': { attributes: { term: 'Album', label: 'Album' } },
            attributes: { term: 'Music', label: 'Music' },
          },
          rights: {
            label:
              'An EMI Records Nashville Release; ℗ 2022 BigEC Records LLC, under exclusive license to UMG Recordings, Inc.',
          },
          title: { label: '& - EP - Eric Church' },
          link: {
            attributes: {
              rel: 'alternate',
              type: 'text/html',
              href: 'https://music.apple.com/us/album/ep/1619175125?uo=2',
            },
          },
          id: {
            label: 'https://music.apple.com/us/album/ep/1619175125?uo=2',
            attributes: { 'im:id': '1619175125' },
          },
          'im:artist': {
            label: 'Eric Church',
            attributes: {
              href: 'https://music.apple.com/us/artist/eric-church/123055194?uo=2',
            },
          },
          category: {
            attributes: {
              'im:id': '6',
              term: 'Country',
              scheme: 'https://music.apple.com/us/genre/music-country/id6?uo=2',
              label: 'Country',
            },
          },
          'im:releaseDate': {
            label: '2022-08-19T00:00:00-07:00',
            attributes: { label: 'August 19, 2022' },
          },
        },
      ],
      updated: { label: '2022-08-23T08:01:59-07:00' },
      rights: { label: 'Copyright 2008 Apple Inc.' },
      title: { label: 'iTunes Store: Top Albums' },
      icon: { label: 'http://itunes.apple.com/favicon.ico' },
      link: [
        {
          attributes: {
            rel: 'alternate',
            type: 'text/html',
            href: 'https://itunes.apple.com/WebObjects/MZStore.woa/wa/viewTop?cc=us&id=1&popId=11',
          },
        },
        {
          attributes: {
            rel: 'self',
            href: 'https://mzstoreservices-int.dslb.apple.com/us/rss/topalbums/limit=4/json',
          },
        },
      ],
      id: {
        label:
          'https://mzstoreservices-int.dslb.apple.com/us/rss/topalbums/limit=4/json',
      },
    },
  };
};
