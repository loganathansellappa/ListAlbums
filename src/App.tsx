import Albums from './components/albums/Albums';
import { Routes, Route, MemoryRouter } from 'react-router-dom';

/**
 * We really dont need to use React Router for current scope as we
 * dont have more pages, however I have added router so we could
 * easily extend the app when we have requirement to add more pages
 */
const App = () => {
  return (
    <MemoryRouter>
      <Routes>
        <Route path="/">
          <Route index element={<Albums />} />
        </Route>
      </Routes>
    </MemoryRouter>
  );
};

export default App;
