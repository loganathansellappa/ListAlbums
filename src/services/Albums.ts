import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { AlbumResponse, Entry } from '../@types/Album';
import { apiService } from '../utils/HelperUtils';

export const albumsApi = createApi({
  reducerPath: 'albumsApi',
  baseQuery: fetchBaseQuery({ baseUrl: apiService() }),
  tagTypes: ['Album'],
  endpoints: (builder) => ({
    getResource: builder.query<Array<Entry>, string>({
      query: (limit) => `limit=${limit}/json`,
      transformResponse: (response: AlbumResponse) => response.feed.entry,
    }),
  }),
});

export const { useGetResourceQuery } = albumsApi;
