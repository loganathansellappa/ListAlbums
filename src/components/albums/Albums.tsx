import React, { useCallback, useEffect, useState } from 'react';
import Error from '../error/Error';
import { useGetResourceQuery } from '../../services/Albums';
import { Loader } from '../loader/Loader';
import { Entry } from '../../@types/Album';
import { Album } from '../album/Album';
import { dataLimit, filteredAlbums } from '../../utils/HelperUtils';
import './Albums.scss';

const Albums: React.FunctionComponent = () => {
  const [search, setSearch] = useState<string>('');
  const [emptySearchResult, setEmptySearchResult] = useState<boolean>(false);
  const updateSearch = useCallback(
    (val: string) => setSearch(val),
    [setSearch]
  );

  const { data, error, isLoading } = useGetResourceQuery(dataLimit(), {
    selectFromResult: ({ data, error, isLoading }) => {
      if (search) data = filteredAlbums(data as Array<Entry>, search);
      return {
        data,
        error,
        isLoading,
      };
    },
  });

  useEffect(() => {
    setEmptySearchResult((search && data?.length === 0) as boolean);
  }, [search, data]);

  if (error) return <Error />;

  return (
    <div className="albums-container">
      <div className="search">
        <input
          className={'searchInput'}
          value={search}
          placeholder="Search Album..."
          onChange={(e) => updateSearch((e.target as HTMLInputElement).value)}
        />
        {emptySearchResult ? (
          <div className={'searchHint'}>
            Album not found. You can search either by Name/Title.
          </div>
        ) : null}
      </div>
      {isLoading ? (
        <Loader className={'inline'} />
      ) : (
        <div className="albums">
          {data?.map((album: Entry) => (
            <Album key={album.id.attributes['im:id']} album={album} />
          ))}
        </div>
      )}
    </div>
  );
};

export default Albums;
