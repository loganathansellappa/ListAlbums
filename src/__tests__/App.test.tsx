import 'whatwg-fetch';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { mockResource, testStore } from '../utils/tests';
import * as Hooks from '../services/Albums';
import App from '../App';

describe('App', () => {
  let element: { container: HTMLElement };

  beforeEach(() => {
    jest.spyOn(Hooks, 'useGetResourceQuery').mockReturnValue({
      data: mockResource().feed.entry,
      isError: false,
      isLoading: false,
      refetch: jest.fn(),
    });
    element = render(
      <Provider store={testStore()}>
        <App />
      </Provider>
    );
  });

  test('renders component', () => {
    expect(element.container.firstChild).toMatchSnapshot();
  });
});
