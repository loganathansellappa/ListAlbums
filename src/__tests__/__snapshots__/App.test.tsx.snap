// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component 1`] = `
<div
  class="albums-container"
>
  <div
    class="search"
  >
    <input
      class="searchInput"
      placeholder="Search Album..."
      value=""
    />
  </div>
  <div
    class="albums"
  >
    <div
      class="card"
    >
      <img
        alt="Life After Death"
        src="https://is2-ssl.mzstatic.com/image/thumb/Music112/v4/5d/1b/af/5d1baf4a-496c-be36-b5c1-ba5bcea1c2c0/22UMGIM57228.rgb.jpg/170x170bb.png"
      />
      <div
        class="name"
      >
        <h3
          class="searchResult"
        >
          Life After Death
        </h3>
      </div>
      <div
        class="propContainer"
      >
        <div
          class="props"
        >
          <div
            class="value"
          >
            Life After Death - TobyMac
          </div>
        </div>
        <div
          class="props"
        >
          <div
            class="label"
          >
            Price
          </div>
          <div
            class="value"
          >
            $10.99
          </div>
        </div>
      </div>
      <div
        class="favourite"
      >
        <div
          class=""
        >
          Mark
           Favourite
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart favouriteIcon"
            color="white"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <img
        alt="Viva Las Vengeance"
        src="https://is5-ssl.mzstatic.com/image/thumb/Music112/v4/bc/49/c6/bc49c6b9-9a12-dd6c-ac50-e4409cf8a094/075679762108.jpg/170x170bb.png"
      />
      <div
        class="name"
      >
        <h3
          class="searchResult"
        >
          Viva Las Vengeance
        </h3>
      </div>
      <div
        class="propContainer"
      >
        <div
          class="props"
        >
          <div
            class="value"
          >
            Viva Las Vengeance - Panic! At the Disco
          </div>
        </div>
        <div
          class="props"
        >
          <div
            class="label"
          >
            Price
          </div>
          <div
            class="value"
          >
            $9.99
          </div>
        </div>
      </div>
      <div
        class="favourite"
      >
        <div
          class=""
        >
          Mark
           Favourite
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart favouriteIcon"
            color="white"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <img
        alt="Finally Enough Love: 50 Number Ones (2022 Remasters)"
        src="https://is1-ssl.mzstatic.com/image/thumb/Music126/v4/f8/70/c2/f870c27d-81f3-08b7-f66d-b7dc3ffe6a1f/603497839797.jpg/170x170bb.png"
      />
      <div
        class="name"
      >
        <h3
          class="searchResult"
        >
          Finally Enough Love: 50 Number Ones (2022 Remasters)
        </h3>
      </div>
      <div
        class="propContainer"
      >
        <div
          class="props"
        >
          <div
            class="value"
          >
            Finally Enough Love: 50 Number Ones (2022 Remasters) - Madonna
          </div>
        </div>
        <div
          class="props"
        >
          <div
            class="label"
          >
            Price
          </div>
          <div
            class="value"
          >
            $24.99
          </div>
        </div>
      </div>
      <div
        class="favourite"
      >
        <div
          class=""
        >
          Mark
           Favourite
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart favouriteIcon"
            color="white"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <img
        alt="& - EP"
        src="https://is4-ssl.mzstatic.com/image/thumb/Music126/v4/88/ba/eb/88baebfa-1338-f4c4-e9ab-a86877a2b763/20UMGIM88356.rgb.jpg/170x170bb.png"
      />
      <div
        class="name"
      >
        <h3
          class="searchResult"
        >
          & - EP
        </h3>
      </div>
      <div
        class="propContainer"
      >
        <div
          class="props"
        >
          <div
            class="value"
          >
            & - EP - Eric Church
          </div>
        </div>
        <div
          class="props"
        >
          <div
            class="label"
          >
            Price
          </div>
          <div
            class="value"
          >
            $6.99
          </div>
        </div>
      </div>
      <div
        class="favourite"
      >
        <div
          class=""
        >
          Mark
           Favourite
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-heart favouriteIcon"
            color="white"
            data-icon="heart"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 190.9V185.1C0 115.2 50.52 55.58 119.4 44.1C164.1 36.51 211.4 51.37 244 84.02L256 96L267.1 84.02C300.6 51.37 347 36.51 392.6 44.1C461.5 55.58 512 115.2 512 185.1V190.9C512 232.4 494.8 272.1 464.4 300.4L283.7 469.1C276.2 476.1 266.3 480 256 480C245.7 480 235.8 476.1 228.3 469.1L47.59 300.4C17.23 272.1 .0003 232.4 .0003 190.9L0 190.9z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
